snapcraft/_store.py

```diff
@@ -623,11 +623,23 @@ def close(snap_name, channel_names):
     logger.info(msg)
 
 
-def download(snap_name, channel, download_path, arch):
-    """Download snap from the store to download_path"""
+def download(snap_name, channel, download_path, arch, but_hash=''):
+    """Download snap from the store to download_path.
+    :param str snap_name: The snap name to download.
+    :param str channel: the channel to get the snap from.
+    :param str download_path: the path to write the downloaded snap to.
+    :param str arch: the architecture of the download as a deb arch.
+    :param str but_hash: a sha3_384 hash to indicate that the download should
+                         not take place.
```

Comment:
This variable name and description don't make sense to me. I don't understand what it does.